### PR TITLE
Deactivate Business Entity codes

### PR DESCRIPTION
**Pull request by colinbruce**
This is the technical implementation of the delete/deactivate feature.
Content updates may need a separate PR when terminology is agreed.

### Replaces #561

### CODE REVIEW

**Comment by dotemacs** [comment]
Looks fine

File: config/routes.rb [hunk 1]
```diff
@@ -85,6 +85,8 @@
 
   resources :offices do
     resources :business_entities
+    get '/business_entities/:id/confirm' => 'business_entities#confirm', as: 'confirm_deactivate_business_entity'
+    post '/business_entities/:id/deactivate' => 'business_entities#deactivate', as: 'deactivate_business_entity'
```
Comment (novotnyjakub) [inline config/routes.rb]: These 2 rules are related to the resource and ideally would be nested in it: 

```ruby
resources :business_entities do
  get :confirm, on: :member
  post :deactivate, on: :member
end
```

Something between these lines.

File: app/controllers/business_entities_controller.rb [hunk 2]
```diff
@@ -27,16 +23,28 @@ def edit
 
   def update
     authorize business_entity_service.build_update(business_entity_params)
+    persist_and_redirect(:edit)
+  end
+
+  def confirm
```
Comment (novotnyjakub) [inline app/controllers/business_entities_controller.rb]: At first sight, this confused me as to what is it confirming? I'm not insisting, but wouldn't it make sense to be `confirm_deactivate` to be clear what the action does?

File: spec/controllers/business_entities_controller_spec.rb [hunk 3]
```diff
@@ -130,4 +130,35 @@
...
+      sign_in admin
+      get :confirm, office_id: office.id, id: business_entity.id
+    end
+
+    it { is_expected.to have_http_status(:success) }
+
+    it { is_expected.to render_template(:confirm) }
+
+    it 'assigns the @business_entity variable' do
+      expect(assigns(:business_entity)).to be_a_kind_of BusinessEntity
```
Comment (novotnyjakub) [inline spec/controllers/business_entities_controller_spec.rb]: Here I think you can test very specifically that `expect(assigns(:business_entity)).to eql(business_entity)`.

File: spec/services/business_entity_service_spec.rb [hunk 4]
```diff
@@ -102,6 +102,24 @@
...
+
+    it 'returns a persisted object' do
+      expect(subject.persisted?).to be true
+    end
+
+    it 'returns a valid object' do
+      expect(subject.valid?).to be true
+    end
+
+    it 'has no ID' do
```
Comment (novotnyjakub) [inline spec/services/business_entity_service_spec.rb]: The test title is different from the test itself.

**Comment by novotnyjakub** [comment]
A few small things, but overall it's nicely build on the previous PRs :+1:

**Pull request merged**